gateway/app/schemas/qodProvisioning.py

```diff
@@ -0,0 +1,219 @@
+]
+
+
+class CredentialType(Enum):
+    PLAIN = "PLAIN"
+    ACCESSTOKEN = "ACCESSTOKEN"
+    REFRESHTOKEN = "REFRESHTOKEN"
+
+
+class SinkCredential(BaseModel):
+    credentialType: Optional[CredentialType] = Field(
+        None, description="The type of the credential."
+    )
+
+
+class PlainCredential(SinkCredential):
+    identifier: str = Field(
+        description="The identifier might be an account or username."
+    )
+    secret: str = Field(description="The secret might be a password or passphrase.")
+
+
+class AccessTokenType(Enum):
+    bearer = "bearer"
+
+
+class AccessTokenCredential(SinkCredential):
+    accessToken: str = Field(
+        description="REQUIRED. An access token is a previously acquired token granting access to the target resource.",
+    )
+    accessTokenExpiresUtc: datetime = Field(
+        description="REQUIRED. An absolute UTC instant at which the token shall be considered expired.",
+    )
+    accessTokenType: AccessTokenType = Field(
+        description="REQUIRED. Type of the access token (See [OAuth 2.0](https://tools.ietf.org/html/rfc6749#section-7.1)).",
+    )
+
+
+class RefreshTokenCredential(SinkCredential):
+    accessToken: str = Field(
+        description="REQUIRED. An access token is a previously acquired token granting access to the target resource.",
+    )
+    accessTokenExpiresUtc: datetime = Field(
+        description="REQUIRED. An absolute UTC instant at which the token shall be considered expired.",
+    )
+    accessTokenType: AccessTokenType = Field(
+        description="REQUIRED. Type of the access token (See [OAuth 2.0](https://tools.ietf.org/html/rfc6749#section-7.1)).",
+    )
+    refreshToken: str = Field(
+        description="REQUIRED. An refresh token credential used to acquire access tokens.",
+    )
+    refreshTokenEndpoint: AnyUrl = Field(
+        description="REQUIRED. A URL at which the refresh token can be traded for an access token.",
+    )
+
+
+class Type(Enum):
+    org_camaraproject_qod_provisioning_v0_status_changed = (
+        "org.camaraproject.qod-provisioning.v0.status-changed"
+    )
+
+
+class Specversion(Enum):
+    field_1_0 = "1.0"
+
+
+class Datacontenttype(Enum):
+    application_json = "application/json"
+
+
+class CloudEvent(BaseModel):
+    id: str = Field(
+        description="Identifier of this event, that must be unique in the source context.",
+    )
+    source: str = Field(
+        description="Identifies the context in which an event happened in the specific Provider Implementation.",
+    )
+    type: Type = Field(description="The type of the event.")
+    specversion: Specversion = Field(
+        description="Version of the specification to which this event conforms (must be 1.0 if it conforms to cloudevents 1.0.2 version)",
+    )
+    datacontenttype: Optional[Datacontenttype] = Field(
+        None,
+        description='media-type that describes the event payload encoding, must be "application/json" for CAMARA APIs',
+    )
```

Comment:
```suggestion
    datacontenttype: Annotated[Optional[Datacontenttype], Field(
        description='media-type that describes the event payload encoding, must be "application/json" for CAMARA APIs',
    )] = None
```

Same as above but notice that the default value is now in the assignment